packages/origin-ui-core/src/components/Modal/WithdrawModal.tsx

```diff
@@ -0,0 +1,120 @@
+import { ProducingDevice } from '@energyweb/device-registry';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    TextField
+} from '@material-ui/core';
+import { bigNumberify } from 'ethers/utils';
+import moment from 'moment';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { resyncCertificate, requestWithdrawCertificate } from '../../features/certificates';
+import { ICertificateViewItem } from '../../features/certificates/types';
+import { getExchangeClient } from '../../features/general/selectors';
+import { getUserOffchain } from '../../features/users/selectors';
+import { EnergyFormatter, formatDate } from '../../utils';
+
+interface IProps {
+    certificate: ICertificateViewItem;
+    producingDevice: ProducingDevice.Entity;
+    showModal: boolean;
+    callback: () => void;
+}
+
+const DEFAULT_ENERGY_IN_BASE_UNIT = bigNumberify(1);
+
+export function WithdrawModal(props: IProps) {
+    const { certificate, callback, producingDevice, showModal } = props;
+
+    const user = useSelector(getUserOffchain);
+
+    const [energyInDisplayUnit, setEnergyInDisplayUnit] = useState(
+        EnergyFormatter.getValueInDisplayUnit(DEFAULT_ENERGY_IN_BASE_UNIT)
+    );
+
+    const dispatch = useDispatch();
+
+    const exchangeClient = useSelector(getExchangeClient);
+
+    useEffect(() => {
+        if (certificate) {
+            setEnergyInDisplayUnit(
+                EnergyFormatter.getValueInDisplayUnit(certificate.energy.publicVolume)
+            );
+        }
+    }, [certificate, user]);
+
+    async function handleClose() {
+        dispatch(resyncCertificate(certificate));
+        callback();
+    }
+
+    async function withdraw() {
+        const account = await exchangeClient.getAccount();
+        const assetId = certificate.assetId;
+        const address = user.blockchainAccountAddress;
+        const amount = account.balances.available.find((balance) => balance.asset.id === assetId)
+            .amount;
+        dispatch(
+            requestWithdrawCertificate({
+                assetId,
+                address,
+                amount,
+                callback
+            })
+        );
+    }
+
+    const certificateId = certificate ? certificate.id : '';
```

Comment:
In general not sure why this code is necessary, in the case that `certificate` object is incorrect you should return and add show error or something.

Reply:
the reason for this is that all component variables must be initialized regardless of whether the component is displayed (and the certificate is selected) or not